lib/puppet/provider/package/pip.rb

```diff
@@ -56,11 +56,8 @@ def query
   # cache of PyPI's package list so this operation will always have to
   # ask the web service.
   def latest
-    client = XMLRPC::Client.new2("http://pypi.python.org/pypi")
-    client.http_header_extra = {"Content-Type" => "text/xml"}
-    client.timeout = 10
-    result = client.call("package_releases", @resource[:name])
-    result.first
+    result = JSON.parse(open("https://pypi.python.org/pypi/#{@resource[:name]}/json").read)
```

Comment:
Ideally you could do something like:

```
http = Puppet::Network::HttpPool.http_instance('pypi.python.org', '443', true)
response = http.get("/pypi/cufflinks/json")
case response
when Net::HTTPSuccess
  puts JSON.parse(response.body)
else
  raise "Failed to request pypi metadata: #{response.code}, #{response.message}"
end
```

However, puppet will only accept the puppet CA resulting in:

```
certificate verify failed: [unable to get local issuer certificate for /C=US/O=DigiCert Inc/OU=www.digicert.com/CN=DigiCert SHA2 Extended Validation Server CA] (Puppet::Error)
```

If you disable ssl verification (for testing):

```
http = Puppet::Network::HttpPool.http_instance('pypi.python.org', '443', true, false)
```

It works as expected:

```
{"info"=>{"maintainer"=>nil, "docs_url"=>nil, "requires_python"=>nil, "maintainer_email"=>nil, "cheesecake_code_kwalitee_id"=>nil, "keywords"=>"pandas,plotly,plotting", "package_url"=>"http://pypi.python.org/pypi/cufflinks", "author"=>"Jorge Santos", "author_email"=>"santos.jorge@gmail.com", "download_url"=>"UNKNOWN", "platform"=>"UNKNOWN", "version"=>"0.6.1", "cheesecake_documentation_id"=>nil, "_pypi_hidden"=>false, "description"=>"UNKNOWN", "release_url"=>"http://pypi.python.org/pypi/cufflinks/0.6.1", "downloads"=>{"last_month"=>2443, "last_week"=>635, "last_day"=>74}, "_pypi_ordering"=>7, "classifiers"=>[], "name"=>"cufflinks", "bugtrack_url"=>nil, "license"=>"MIT", "summary"=>"Productivity Tools for Plotly + Pandas", "home_page"=>"https://github.com/santosjorge/cufflinks", "stable_version"=>nil, "cheesecake_installability_id"=>nil}, "releases"=>{"0.3.2"=>[{"has_sig"=>false, "upload_time"=>"2015-06-08T11:52:13", "comment_text"=>"", "python_version"=>"source", "url"=>"https://pypi.python.org/packages/source/c/cufflinks/cufflinks-0.3.2.tar.gz", "md5_digest"=>"818dd8967156698fdc8b62da02900aab", "downloads"=>3050, "filename"=>"cufflinks-0.3.2.tar.gz", "packagetype"=>"sdist", "size"=>15136}], "0.6.1"=>[{"has_sig"=>false, "upload_time"=>"2015-08-17T20:54:05", "comment_text"=>"", "python_version"=>"source", "url"=>"https://pypi.python.org/packages/source/c/cufflinks/cufflinks-0.6.1.tar.gz", "md5_digest"=>"19bd13762a5b06d1c115eb03b7d7397d", "downloads"=>0, "filename"=>"cufflinks-0.6.1.tar.gz", "packagetype"=>"sdist", "size"=>28528}], "0.2"=>[{"has_sig"=>false, "upload_time"=>"2015-02-14T19:18:40", "comment_text"=>"", "python_version"=>"source", "url"=>"https://pypi.python.org/packages/source/c/cufflinks/cufflinks-0.2.tar.gz", "md5_digest"=>"68e60ef6f95c83d9b9be14f1bb345d52", "downloads"=>612, "filename"=>"cufflinks-0.2.tar.gz", "packagetype"=>"sdist", "size"=>6932}], "0.5"=>[{"has_sig"=>false, "upload_time"=>"2015-07-24T19:55:46", "comment_text"=>"", "python_version"=>"source", "url"=>"https://pypi.python.org/packages/source/c/cufflinks/cufflinks-0.5.tar.gz", "md5_digest"=>"0667054349dc75031a324cab641cc7f2", "downloads"=>393, "filename"=>"cufflinks-0.5.tar.gz", "packagetype"=>"sdist", "size"=>22902}], "0.4"=>[{"has_sig"=>false, "upload_time"=>"2015-07-18T21:09:10", "comment_text"=>"", "python_version"=>"source", "url"=>"https://pypi.python.org/packages/source/c/cufflinks/cufflinks-0.4.tar.gz", "md5_digest"=>"d389e2ecbe655bf3bfd538f4176240c2", "downloads"=>322, "filename"=>"cufflinks-0.4.tar.gz", "packagetype"=>"sdist", "size"=>21465}], "0.6"=>[{"has_sig"=>false, "upload_time"=>"2015-08-17T20:19:48", "comment_text"=>"", "python_version"=>"source", "url"=>"https://pypi.python.org/packages/source/c/cufflinks/cufflinks-0.6.tar.gz", "md5_digest"=>"813b4c9842e483d48870945d38dfee5f", "downloads"=>0, "filename"=>"cufflinks-0.6.tar.gz", "packagetype"=>"sdist", "size"=>28499}], "0.1"=>[], "0.3"=>[{"has_sig"=>false, "upload_time"=>"2015-05-05T01:38:10", "comment_text"=>"", "python_version"=>"source", "url"=>"https://pypi.python.org/packages/source/c/cufflinks/cufflinks-0.3.tar.gz", "md5_digest"=>"bcfa36336c29170a31b4dc7c3928e4ba", "downloads"=>946, "filename"=>"cufflinks-0.3.tar.gz", "packagetype"=>"sdist", "size"=>13552}]}, "urls"=>[{"has_sig"=>false, "upload_time"=>"2015-08-17T20:54:05", "comment_text"=>"", "python_version"=>"source", "url"=>"https://pypi.python.org/packages/source/c/cufflinks/cufflinks-0.6.1.tar.gz", "md5_digest"=>"19bd13762a5b06d1c115eb03b7d7397d", "downloads"=>0, "filename"=>"cufflinks-0.6.1.tar.gz", "packagetype"=>"sdist", "size"=>28528}]}
```

The trouble seems to be that puppet's http API doesn't allow the caller to specify they want to accept the default set of cacerts. To do that you need access to the underlying ruby connection `Net::HTTP` so you can do:

```
http.cert_store.set_default_paths
```

But callers to puppet API don't have access to the `Net::HTTP` object, as it's only in scope once we've created a new connection (or borrowed one from the pool).

We should probably file a ticket to make that possible, so that the call to `OpenSSL::X509::Store#set_default_paths` happens automatically.